src/Service.js

```diff
@@ -75,6 +75,47 @@ class Service {
   }
 
   /**
+   * Retrieves the array of task info objects available on the service.
+   * @return {Promise<TaskInfo[], error>} Returns a Promise to an array of TaskInfo objects.
+   */
+  taskInfoList() {
+    return new Promise((resolve, reject) => {
+      // Build service info url.
+      const url = [this._server.rootURL, SERVER_API.SERVICES_PATH, this.name].join('/');
+
+      // Get service info so we can pull off the tasks array.
+      request
+        .get(url)
+        .query({ taskInfo: true })
+        .use(nocache) // Prevents caching of *only* this request
+        .end((err, res) => {
+          if (res && res.ok) {
+            const tasks = [];
+            res.body.tasks.forEach((task) => {
+              if (typeof task === 'string') {
```

Comment:
Just curious, why is this check important? Are there any other types that could mean error?